src/Npgsql/Replication/ReplicationConnection.cs

```diff
@@ -444,6 +444,28 @@ async Task<TimelineHistoryFile> TimelineHistoryInternal(uint tli, CancellationTo
         }
     }
 
+    internal async Task<PhysicalReplicationSlot?> ReadReplicationSlotInternal(string slotName, CancellationToken cancellationToken = default)
+    {
+        var result = await ReadSingleRow($"READ_REPLICATION_SLOT {slotName}", cancellationToken);
+        var slotType = (string?)result[0];
+
+        // Currently (2021-12-30) slot_type is always 'physical' for existing slots or null for slot names that don't exist but that
+        // might change and we'd have to adopt our implementation in that case so check it just in case
+        switch (slotType)
+        {
+            case "physical":
+                var restartLsn = (string?)result[1];
+                var restartTli = (ulong?)result[2];
+                return new PhysicalReplicationSlot(slotName.ToLowerInvariant(),
+                    restartLsn == null ? null : NpgsqlLogSequenceNumber.Parse(restartLsn), restartTli);
```

Comment:
Tiny nit suggestion (when putting arguments on separate lines I tend to do them all, possible EF influence):

```suggestion
                return new PhysicalReplicationSlot(
                    slotName.ToLowerInvariant(),
                    restartLsn == null ? null : NpgsqlLogSequenceNumber.Parse(restartLsn),
                    restartTli);
```

Reply:
Yeah, I see no problem.
I tend to accept whatever Rider makes out of long lines but I can try to remember that.